compysition/actors/eventattributemodifier.py

```diff
@@ -177,4 +177,141 @@ def get_modify_value(self, event):
+        clazz = self.error_class if self.error_class else clazz
+
+        event._set_error(clazz(message=message, code=code, override=override))
+        self.send_event(event)
+
+    def _get_message(self, event):
+        return (self.message if self.message else event.data_string())
+
+    def _get_code(self, event):
+        return (self.code if self.code else None)
+
+class XMLErrorEventAttributeModifier(ErrorEventAttributeModifier):
+    '''
+    Description:
+        Allows for custom errors as they pertain to incoming XML event.data.
+
+    Uses:
+        - Adjusting error (message, code) staticly
+        - Adding http error response format (override)
+        - Adjusting error class
+
+    Example XML input:
+        <data>
+            <response>
+                <response_message>Something went wrong.</response_message>
+                <response_code>6768<response_code>
+                <response_id>1234<response_id>
+            </response>
+        </data>
+
+    Parameters:
+
+        - name:                                             (See Actor)
+        - override (bool):                                  (See ErrorEventAttributeModifier)
+        - code (str):                                       (See ErrorEventAttributeModifier)
+        - code_attr (str)                                   Specifies the variable from an incoming event.data XML object to be used as the new error code.
+                                                                Attribute specified by a '.' seperated attribute list.  (EX: "data.response.response_code")
+                                                                If the variable is missing then a static code is used if provided.
+        - message (str):                                    (See ErrorEventAttributeModifier)
+        - message_attr (str):                               Specifies the variable from an incoming event.data XML object to be used as the new error message.
+                                                                Attribute specified by a '.' seperated attribute list.  (EX: "data.response.response_message")
+                                                                If the variable is missing then a static message is used if provided.
+        - error_class (CompysitionError class):             (See ErrorEventAttributeModifier)
+        - default_error_class (CompysitionError class):     (See ErrorEventAttributeModifier)
+    '''
+    def __init__(self, name, override=False, code=None, code_attr=None, message=None, message_attr=None, error_class=None, default_error_class=MalformedEventData, *args, **kwargs):
+        self.message_attr = message_attr
+        self.code_attr = code_attr
+        super(XMLErrorEventAttributeModifier, self).__init__(name=name, override=override, code=code, message=message, error_class=error_class, default_error_class=default_error_class, *args, **kwargs)
+
+    def _get_attr(self, event, attr):
```

Comment:
Would this maybe be more efficient converting to a dict using `json` from the standard lib?